lib/greek_stemmer.rb

```diff
@@ -21,8 +21,11 @@ def load_settings(key)
     end
   end
 
+
   # Transformations for step 1 words
   STEP_1_EXCEPTIONS = load_settings("step_1_exceptions")
+  # Regex Matches terms that have with step 1 exceptions as suffix
+  STEP_1_REGEX = /(.*)(#{STEP_1_EXCEPTIONS.keys.join("|")})$/u
```

Comment:
This is mostly a style thing but I would also think of hiding this under a method and memoizing the result, e.g.

```ruby
def step_1_regex
  @step_1_regex ||= /(.*)(#{STEP_1_EXCEPTIONS.keys.join("|")})$/u
end
```